File: src/main/java/fr/raconteur/simpleskinswapper/networking/HandshakePayload.java
package fr.raconteur.simpleskinswapper.networking;

import net.minecraft.network.protocol.common.custom.CustomPacketPayload;
import net.minecraft.resources.Identifier;

public record HandshakePayload() implements CustomPacketPayload {
    public static final HandshakePayload INSTANCE = new HandshakePayload();
    public static final CustomPacketPayload.Type<HandshakePayload> PACKET_ID =
            new CustomPacketPayload.Type<>(Identifier.fromNamespaceAndPath("skinshuffle", "handshake"));

    @Override
    public Type<? extends CustomPacketPayload> type() {
        return PACKET_ID;
    }
}

File: src/main/kotlin/fr/raconteur/simpleskinswapper/networking/HandshakePayload.kt
package fr.raconteur.simpleskinswapper.networking

import net.minecraft.network.protocol.common.custom.CustomPacketPayload
import net.minecraft.resources.Identifier

object HandshakePayload : CustomPacketPayload {
    @JvmField
    val PACKET_ID: CustomPacketPayload.Type<HandshakePayload> =
        CustomPacketPayload.Type(Identifier.fromNamespaceAndPath("skinshuffle", "handshake"))

    override fun type(): CustomPacketPayload.Type<out CustomPacketPayload> = PACKET_ID
}
